src/dmtcp_coordinator.cpp

```diff
@@ -891,14 +891,27 @@ DmtcpCoordinator::processDmtUserCmd(DmtcpMessage &hello_remote,
   }
 }
 
+/*
+ * Returns the current timestamp with nanosecond resolution
+ */
+static uint64_t
+getCurrTimestamp()
+{
+  struct timespec value;
+  uint64_t nsecs = 0;
+  JASSERT(clock_gettime(CLOCK_MONOTONIC, &value) == 0);
+  nsecs = value.tv_sec*100000000L + value.tv_nsec;
```

Comment:
Good catch! I have updated the PR now and fixed a similar bug in `UniquePid::ThisProcess()` where we were using the same code.